### ISSUE TITLE
Error responses are not stringified correctly

### ISSUE THREAD

**Issue by delvedor** [fastify/fastify#85]
Use case:
```js
fastify.addHook('preHandler', (req, reply, next) => {
  if (someCondition) {
    return next(new Error('some error'))
  }
}
```
If the error case is triggered, the client will receive `{}`, because `JSON.stringify` does not stringify the error properties.
This can be easily fixed by doing the following:
```js
JSON.stringify(new Error('some error', ['message', 'arguments', 'type', 'name']))
// => "{"message":"some error","name":"Error"}"
```

The `reply` [implementation](https://github.com/fastify/fastify/blob/master/lib/reply.js#L39-L47) of the error handling uses [fast-json-stringify](https://github.com/davidmarkclements/fast-safe-stringify) that does not support *replacer* functions, so we can make a PR to fast-json-stringify or move to JSON.stringify for the errors.

**Comment by delvedor** [comment]
cc @mcollina @davidmarkclements

**Comment by delvedor** [comment]
We found that this can be easily fixed by adopting [Boom](https://www.npmjs.com/package/boom), as said in https://github.com/fastify/fastify/issues/69.

**Comment by delvedor** [comment]
Closed by #86
